src/components/Focus/Focus.tsx

```diff
@@ -4,7 +4,7 @@ import {focusFirstFocusableNode} from '@shopify/javascript-utilities/focus';
 export interface FocusProps {
   children?: React.ReactNode;
   disabled?: boolean;
-  root: HTMLElement | null;
+  root: React.RefObject<HTMLElement> | HTMLElement | null;
```

Comment:
I feel like passing the entire ref is safer, but this would break the API of this component - would it make sense to go towards passing the entire ref and add a warning of deprecation for passing only a current element?

Reply:
I'm a fan of this and would prefer to remove `HTMLElement` altogether but that can be done in another PR since it would be a breaking change.

Reply:
Because it's a breaking change, is there a way in Polaris to have a warning for future deprecations?